src/EventGridEmulator/EventHandling/PullQueueHttpContextHandler.cs

```diff
@@ -0,0 +1,142 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+using Azure.Messaging;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EventGridEmulator.EventHandling;
+
+internal sealed class PullQueueHttpContextHandler
+{
+    [StringSyntax("Route")]
+    public const string ReceiveRoute = "topics/{topic}/eventsubscriptions/{subscription}:receive";
+
+    [StringSyntax("Route")]
+    public const string AcknowledgeRoute = "topics/{topic}/eventsubscriptions/{subscription}:acknowledge";
+
+    [StringSyntax("Route")]
+    public const string ReleaseRoute = "topics/{topic}/eventsubscriptions/{subscription}:release";
+
+    [StringSyntax("Route")]
+    public const string RejectRoute = "topics/{topic}/eventsubscriptions/{subscription}:reject";
+
+    public static async Task<IResult> HandleReceiveAsync([FromRoute] string topic, [FromRoute] string subscription, [FromServices] TopicSubscribers<CloudEvent> events, CancellationToken cancellationToken)
+    {
+        var result = await events.GetEventAsync(topic, subscription, cancellationToken);
+        return Results.Ok(new { value = new[] { new { brokerProperties = new { deliveryCount = 1, lockToken = result.LockToken }, @event = result.Item } } });
```

Comment:
Is this type available in the EventGrid SDK? If yes it could be easier to read than an anynomous type

Reply:
Given they are unavailable, created our own internal classes